### ISSUE TITLE
LookupError: no codec search functions registered: can't find encoding

### ISSUE THREAD

**Issue by aeromaxx** [ytdl-org/youtube-dl#29177]
## Checklist

- [x] I'm asking a question
- [x] I've looked through the README and FAQ for similar questions
- [x] I've searched the bugtracker for similar questions including closed ones

## Question

I've installed the latest version of youtube-dl on a Debian Stretch headless server, and I'm getting a LookupError message stating "LookupError: no codec search functions registered: can't find encoding".

I'm confused as to what they actually means, youtube-dl has been installed on this server a while but only just now have I noticed getting this error, so am unsure as to when it started happening.

You youtube-dl wasn't install via apt-get install youtube-dl, strangely enough the version installed via apt works, but the one installed from the github repo doesn't.

Would someone be able to explain the error maybe? and the steps I need to take to fix it?

**Comment by rautamiekka** [comment]
Fill the template fully, it's impossible to help with only this info.

Since you're on Debian, don't use APT to install the app unless you're using an automated repo (which to my knowledge doesn't exist).

**Comment by aeromaxx** [comment]
> Fill the template fully, it's impossible to help with only this info.

I have filled in the template fully, I used the question template and it only asked me to write my question which is what I have done.

> Since you're on Debian, don't use APT to install the app unless you're using an automated repo (which to my knowledge doesn't exist).

I didn't use APT to install the app, I did this to test if the one in the debian repo worked and it does, I then uninstalled it.  Installed the one from github and that one gives me the errors mentioned in my initial question.
